applications/GeoMechanicsApplication/custom_elements/transient_thermal_element.h

```diff
@@ -86,6 +94,22 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) TransientThermalElement : public Ele
 
     GeometryData::IntegrationMethod GetIntegrationMethod() const override
     {
+        if (GetGeometry().LocalSpaceDimension() == 1)
+        {
+            switch (TNumNodes) {
+            case 2:
+            case 3:
+                return GeometryData::IntegrationMethod::GI_GAUSS_2;
+            case 4:
+                return GeometryData::IntegrationMethod::GI_GAUSS_3;
+            case 5:
+                return GeometryData::IntegrationMethod::GI_GAUSS_5;
+            default:
+                KRATOS_ERROR << "Can't return integration method: unexpected number of nodes: " << TNumNodes
+                             << std::endl;
+            }
+        }
```

Comment:
This switch statement is the same for the thermal element and the 1d pw element. I would suggest to extract a utility function for this for the PR that goes back latest. I would suggest to try and first merge the thermal element and afterwards the Pw, since this one is way smaller and more straight-forward